server/src/discord/discordApi.ts

```diff
@@ -189,6 +189,17 @@ ${await teamsSection()}${await usersSection()}
       console.log(`Failed to update LFG post: ${e}`)
     }
   }
+  
+  async resetTeamDefaultChannel(targetTeam: TeamT) {
```

Comment:
Similar to how hwApi works off the types defined in the app, *this* file only works with discord "types".  Imagine if this were shared with other users as an API to access discord APIs. If you have a tic-tac-toe game, there's no concept of a 'TeamT'. Also this API shouldn't be calling things inside the firebaseAPI (I did break this rule for the LFG/LFM stuff, which perhaps should be changed as well, but it's fine to leave that part alone for your change)

I think if this file is exposing discord functionality, it's fine to have a 'recreateChannel' function that gets the category, etc, and then just returns the new channel ID, which is saved at the call site for this function, presumably in hwApi.

Reply:
agreed I will update the params

about second point u mentioned `file`...iiuc `recreateChannel` should be a static method of discordApi, right?

Reply:
There aren't any static methods, it should just be a method in discordApi like any other, that takes in a channel id, and returns the newly created channel id